### ISSUE TITLE
How do I load a pre-trained word embedding model into an sentence transformer built from scratch?

### ISSUE THREAD

**Issue by jplasser** [UKPLab/sentence-transformers#1300]
I have the following code where I am trying to create a model that matches the following line of code, but built from _scratch_:
```Python
model = SentenceTransformer('all-mpnet-base-v2')
```
I then programmed the following lines and found that the pre-trained model weights were not being loaded.

```Python
from sentence_transformers import SentenceTransformer, models
from torch import nn

# creating different layers
word_embedding_model = models.Transformer('microsoft/mpnet-base', max_seq_length=384)
pooling_model = models.Pooling(word_embedding_model.get_word_embedding_dimension())
dense_model = models.Dense(in_features=pooling_model.get_sentence_embedding_dimension(), out_features=1024,
                           activation_function=nn.Tanh())
normalize_model = models.Normalize()

# build the final model
model = SentenceTransformer(modules=[word_embedding_model, pooling_model, dense_model, normalize_model])

# freeze transformer layers of the model
auto_model = model._first_module().auto_model
for param in auto_model.parameters():
    param.requires_grad = False
```
How can I load the pre-trained weights into this model, as with "all-mpnet-base-v2" above?
Or more generally, how can I use any pre-trained Transformer model like BERT and then fine tune a sentence transformer with it?

**Comment by jplasser** [comment]
Good! If I understand it correctly the only difference of instantiating the model as in `model = SentenceTransformer('all-mpnet-base-v2')` to building it from scratch is that only the word_embeddings are pretrained in the latter case and the pooling layer is not (an additional dense layer is also not), while in the former case (`model = SentenceTransformer('all-mpnet-base-v2')`) all weights have been pretrained accordingly and are loaded on instantiation?

Why do I ask? Because I see different results in inference compared to the pretrained model, when I build the model from scratch (even without the dense layer):

```Python
# model from scratch
from sentence_transformers import SentenceTransformer, models
from torch import nn

word_embedding_model = models.Transformer('microsoft/mpnet-base', max_seq_length=384)
pooling_model = models.Pooling(word_embedding_model.get_word_embedding_dimension())

normalize_model = models.Normalize()
model = SentenceTransformer(modules=[word_embedding_model, pooling_model, normalize_model])
```

Shouldn't the results be equal? Is the pooling layer trainable? This could be a reason for different results, as I only load the pretrained weights for the transformer.

Here's a heatmap of the inner product of the encoder vectors from `model = SentenceTransformer('all-mpnet-base-v2')`:
![image](https://user-images.githubusercontent.com/4097239/145005232-e5a6b903-c8b8-48aa-aa42-6d455485bf6c.png)

Compare this to the model built from scratch (coded as above), just doing inference, no fine tuning or any training done:
![image](https://user-images.githubusercontent.com/4097239/145005398-bfa44b3f-84e9-4e98-a6c0-2cf75b1572be.png)

Is this as expected?

**Comment by nreimers** [comment]
In the above code you use 'microsoft/mpnet-base', which was not trained for text embeddings.

Replace that with 'sentence-transformers/all-mpnet-base-v2' and you will get the same results.

Pooling layer has no parameters. Normalization layer also has no parameters. Dense layer has parameters, so if you add it from scratch, it will be initialized randomly.

**Comment by jplasser** [comment]
Thank you for your reply, now it is clear!
